src/test/check_getdns_context_set_dns_transport.h

```diff
@@ -71,108 +71,222 @@
 
+       /* Request a response that should be truncated over UDP */
        ASSERT_RC(getdns_context_set_dns_transport(context, GETDNS_TRANSPORT_UDP_ONLY),
-         GETDNS_RETURN_GOOD, "Return code from getdns_context_set_dns_transport()");   
-     
+         GETDNS_RETURN_GOOD, "Return code from getdns_context_set_dns_transport()");
+       ASSERT_RC(getdns_context_set_edns_maximum_udp_payload_size(context, 512),
+           GETDNS_RETURN_GOOD, "Return code from getdns_context_set_edns_maximum_udp_payload_size()"); 
+       ASSERT_RC(getdns_context_set_edns_do_bit(context, 1),
+           GETDNS_RETURN_GOOD, "Return code from getdns_context_set_edns_do_bit()");
 
-       ASSERT_RC(getdns_general_sync(context, "google.com", 255, NULL, &response), 
+       ASSERT_RC(getdns_general_sync(context, "getdnsapi.net", 48, extensions, &response), 
          GETDNS_RETURN_GOOD, "Return code from getdns_general_sync()");
 
-       EXTRACT_RESPONSE;
+       ASSERT_RC(getdns_dict_get_int(response, "/call_reporting/0/transport", &transport),
+         GETDNS_RETURN_GOOD, "Failed to extract \"transport\"");
+       ASSERT_RC(transport, GETDNS_TRANSPORT_UDP, "Query did not go over UDP");
+       ASSERT_RC(getdns_dict_get_int(response, "/replies_tree/0/header/tc", &tc),
+         GETDNS_RETURN_GOOD, "Failed to extract \"tc\"");
+       ASSERT_RC(tc, 1, "Packet not trucated as expected");
 
+       /* Re-do over TCP */
+       ASSERT_RC(getdns_context_set_dns_transport(context, GETDNS_TRANSPORT_TCP_ONLY),
+         GETDNS_RETURN_GOOD, "Return code from getdns_context_set_dns_transport()");   
 
-      ASSERT_RC(getdns_dict_get_int(ex_response.header, "ancount", &ancount),
-        GETDNS_RETURN_GOOD, "Failed to extract \"nscount\"");
+       ASSERT_RC(getdns_general_sync(context, "getdnsapi.net", 48, extensions, &response), 
+         GETDNS_RETURN_GOOD, "Return code from getdns_general_sync()");
 
-      ASSERT_RC(getdns_dict_get_int(ex_response.header, "arcount", &arcount),
-        GETDNS_RETURN_GOOD, "Failed to extract \"nscount\"");
+       ASSERT_RC(getdns_dict_get_int(response, "/call_reporting/0/transport", &transport),
+         GETDNS_RETURN_GOOD, "Failed to extract \"transport\"");
+       ASSERT_RC(transport, GETDNS_TRANSPORT_TCP, "Query did not go over TCP");
+       ASSERT_RC(getdns_dict_get_int(response, "/replies_tree/0/header/tc", &tc),
+         GETDNS_RETURN_GOOD, "Failed to extract \"tc\"");
+       ASSERT_RC(tc, 0, "Packet trucated - not as expected");
 
-      ASSERT_RC(getdns_dict_get_int(ex_response.header, "nscount", &nscount),
-        GETDNS_RETURN_GOOD, "Failed to extract \"nscount\"");
+       /* Now let it fall back to TCP */
+       ASSERT_RC(getdns_context_set_dns_transport(context, GETDNS_TRANSPORT_UDP_FIRST_AND_FALL_BACK_TO_TCP),
+         GETDNS_RETURN_GOOD, "Return code from getdns_context_set_dns_transport()");   
+       ASSERT_RC(getdns_general_sync(context, "getdnsapi.net", 48, extensions, &response), 
```

Comment:
Why not use the constant name GETDNS_RRTYPE_DNSKEY?  In this case using a macro would improve readability.